File: Api/Models/Hospital.cs
using Api.Models.Common;

namespace Api.Models;

public sealed class Hospital : AuditableEntity
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string PostalCode { get; set; } = string.Empty;
    public string Street { get; set; } = string.Empty;
    public string Number { get; set; } = string.Empty;
    public string? Complement { get; set; }
    public string Neighborhood { get; set; } = string.Empty;
    public string City { get; set; } = "Araras";
    public string State { get; set; } = "SP";
    public decimal Latitude { get; set; }
    public decimal Longitude { get; set; }
    public bool HasEmergencyDepartment { get; set; } = true;
    public bool Active { get; set; } = true;
    public ICollection<HospitalWard> Wards { get; set; } = new List<HospitalWard>();
    public ICollection<User> Users { get; set; } = new List<User>();
    public ICollection<PatientTransport> Transports { get; set; } = new List<PatientTransport>();
}

File: WebApp/src/interfaces/hospitalInterfaces/Hospital.ts
export interface HospitalWard {
  id: number; hospitalId: number; name: string; totalBeds: number; occupiedBeds: number;
  reservedBeds: number; availableBeds: number; active: boolean; updatedAt: string;
}
export interface Hospital {
  id: number; name: string; postalCode: string; street: string; number: string; complement?: string;
  neighborhood: string; city: string; state: string; latitude: number; longitude: number;
  hasEmergencyDepartment: boolean; active: boolean; wards: HospitalWard[]; updatedAt: string;
}
export interface HospitalPayload extends Omit<Hospital, 'id' | 'wards' | 'updatedAt'> { expectedUpdatedAt?: string }
export interface PatientTransport {
  id: number; occurrenceId: number; requestedByDispatchId: number; samuDispatchId?: number; samuUnitId?: number;
  hospitalId?: number; hospitalWardId?: number; hospitalName?: string; wardName?: string;
  status: 'AGUARDANDO_CENTRAL' | 'AGUARDANDO_DESTINO' | 'HOSPITAL_AVISADO' | 'EM_TRANSPORTE' | 'RECEBIDO' | 'CANCELADO';
  operationalNotes?: string; priority?: string; type?: string; description: string; origin: string;
  estimatedMinutes?: number; acknowledgedAt?: string; createdAt: string; updatedAt: string;
}
